client/post-editor/post-editor.jsx

```diff
@@ -423,12 +423,7 @@ export const PostEditor = React.createClass( {
 		// Some browsers, as Firefox, return a DOMStringList instead, see
 		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
 		const orgTypes = e.clipboardData.types;
-		let types = [];
-		if ( orgTypes.contains ) {
-			types = Array.from( orgTypes );
-		} else {
-			types = orgTypes;
-		}
+		const types = orgTypes.contains ? Array.from( orgTypes ) : orgTypes;
```

Comment:
Not necessarily the scope of this PR (we like higher merge velocity :]), but it could make sense to:

- separate the clipboard processing from the Google Docs-specific testing;
- perform the clipboard processing before passing data to the Docs tester.

Roughly speaking:

```js
onPasteEvent( e ) {
  const clipboard = parseClipboardData( e );
  if ( hasBeenCopiedFromDocs( clipboard ) ) {
    this.props.pasteEventFromDocs();
  }
}
```

This should help make `hasTextBeenCopiedFromGoogleDocs` small enough that `isGoogleDocType` can be inlined. Maybe somewhere like `lib/paste-events` could accommodate a very simple function that performs the type casting you've done here. It's a suggestion that I don't feel strongly about, and one that doesn't concern this PR. :)

Secondly, `hasTextBeenCopiedFromGoogleDocs` and `isGoogleDocType` don't technically need to be methods of `PostEditor`. Not only do they not require access to the instance, but they are more removed from the component's general logic (e.g. processing state, props, or facilitating rendering). In fact, they are tiny pure functions of very specific types (one takes a `ClipboardEvent` or derived format, the other takes a plain string). This one is quite subjective, but serves as food for thought that those functions could be pulled from another module.